### ISSUE TITLE
is this client stable? ValueTransferTransaction fail

### ISSUE THREAD

**Issue by ysykzheng** [klaytn/caver-java#134]
My project want try use this client, version below:
```
<dependency>
      <artifactId>core</artifactId>
      <groupId>com.klaytn.caver</groupId>
      <version>1.0.2</version>
    </dependency>
```

but with below code for transaction fail.

```
Caver caver = Caver.build(
        new HttpService("https://api.cypress.klaytn.net:8651/", new OkHttpClient.Builder().build(),
            false));
    KlayCredentials credentials = KlayCredentials
        .create("private key");
    TransactionManager transactionManager = new TransactionManager.Builder(caver, credentials)
        .setChaindId(ChainId.MAINNET).build();

    ValueTransferTransaction valueTransferTransaction = ValueTransferTransaction.create(
        credentials.getAddress(),  // fromAddress
        "toAddress",  // toAddress
        Convert.toPeb(BigDecimal.ONE, Convert.Unit.PEB).toBigInteger(),  // value
        BigInteger.valueOf(100_000)  // gasLimit
    );
    ;
    KlayRawTransaction klayRawTransaction = transactionManager.sign(valueTransferTransaction);
    String rawTransaction = klayRawTransaction.getValueAsString();
    System.out.println(rawTransaction);


Bytes32 transactionHash = caver.klay().sendSignedTransaction(rawTransaction).send();
    if (transactionHash.hasError()) {
      Error error = transactionHash.getError();
      System.out.println(error.getCode());
      System.out.println(error.getMessage());
      System.out.println(error.getData());
    } else {
      System.out.println(transactionHash.getResult());
    }
```

and i also try use ValueTransfer, but fail.
```
KlayTransactionReceipt.TransactionReceipt transactionReceipt = ValueTransfer
        .create(caver, credentials, ChainId.MAINNET)
        .sendFunds(ValueTransferTransaction.create(
            credentials.getAddress(),  // fromAddress
            "toAddress",  // toAddress
            Convert.toPeb(BigDecimal.ONE, Unit.PEB).toBigInteger(),  // value
            BigInteger.valueOf(100_000)  // gasLimit
        )).send();
```

**Comment by sirano11** [comment]
Hi, ysykzheng.
Would you give me a more information? 
  - Failed HTTP Log
  - Failed transaction's Hash

**Comment by ysykzheng** [comment]
```
String rawTransaction = "0x08f87f1a8505d21dba00830186a094e0711069b41397827fdfc3f31c901018e4301500019480f4009fb84eb6a151565657fb272bb9f016efb6f847f845824055a0ab0871aced6f1a73b55bf0491cb0039b703eac64dd57745e5404218e64cd6ce3a0588f3150f4096ab697fc59878a084d29599faf94bc72658dd5a15332a83cdb5d";
Bytes32 transactionHash = caver.klay().sendSignedTransaction(rawTransaction).send();
    if (transactionHash.hasError()) {
      Error error = transactionHash.getError();
      System.out.println(error.getCode());
      System.out.println(error.getMessage());
      System.out.println(error.getData());
    } else {
      System.out.println(transactionHash.getResult());
    }
```

output:
```
-32000
invalid transaction v, r, s values of the sender
null
```

**Comment by sirano11** [comment]
Have you ever updated through your credential AccountUpdate Transaction?

**Comment by ysykzheng** [comment]
i copy code from this docs https://docs.klaytn.com/bapp/sdk/caver-java/getting-started
I do not understand what you said? How to do it, can you give an example?

**Comment by sirano11** [comment]
This error occurs because the signature generated from your private key is invalid. That's why i asked, if you update your account through AccountUpdate, you must to sign it with the updated key.

If you are curious about AccountUpdate, please refer to the link below.
https://docs.klaytn.com/bapp/sdk/caver-java/getting-started#account-update

Also, I tried running ValueTransfer tranasction on the Baobab testnet with the code you provided, and it worked fine. Attach the code I tested.

I would appreciate if you confirm once again that the Private Key you used to sign is the correct Private Key that corresponds to the From Address.

```
public void testValueTransfer() {
        String privateKey = "your private key";
        String toAddress = "toAddress";
        Caver caver = Caver.build(Caver.BAOBAB_URL);

        KlayCredentials credentials = KlayCredentials.create(privateKey);
        TransactionManager transactionManager = new TransactionManager.Builder(caver, credentials)
                .setChaindId(ChainId.BAOBAB_TESTNET)
                .build();

        ValueTransferTransaction transferTransaction = ValueTransferTransaction.create(
                credentials.getAddress(),
                toAddress, // to Address
                Convert.toPeb(BigDecimal.ONE, Convert.Unit.PEB).toBigInteger(),
                BigInteger.valueOf(100_000));

        KlayRawTransaction klayRawTransaction = transactionManager.sign(transferTransaction);

        String rawTransaction = klayRawTransaction.getValueAsString();
        System.out.println(rawTransaction);

        Bytes32 transactionHash = caver.klay().sendSignedTransaction(rawTransaction).send();
        if (transactionHash.hasError()) {
            Response.Error error = transactionHash.getError();
            System.out.println(error.getCode());
            System.out.println(error.getMessage());
            System.out.println(error.getData());
        } else {
            System.out.println("TransactionHash : " + transactionHash.getResult());
        }

        TransactionReceiptProcessor receiptProcessor = new PollingTransactionReceiptProcessor(caver, 1000, 15);
        KlayTransactionReceipt.TransactionReceipt transactionReceipt = receiptProcessor.waitForTransactionReceipt(transactionHash.getResult());

        System.out.println("TransactionReceipt : " + transactionReceipt);
}

public void testValueTransfer2() {
        String privateKey = "your private key";
        String toAddress = "toAddress";
        Caver caver = Caver.build(Caver.BAOBAB_URL);

        KlayCredentials credentials = KlayCredentials.create(privateKey);

        KlayTransactionReceipt.TransactionReceipt transactionReceipt = ValueTransfer
                .create(caver, credentials, ChainId.BAOBAB_TESTNET)
                .sendFunds(ValueTransferTransaction.create(
                        credentials.getAddress(),  // fromAddress
                        toAddress,  // toAddress
                        Convert.toPeb(BigDecimal.ONE, Convert.Unit.PEB).toBigInteger(),  // value
                        BigInteger.valueOf(100_000)  // gasLimit
                )).send();
}
```

**Comment by ysykzheng** [comment]
hello.

> This error occurs because the signature generated from your private key is invalid. That's why i asked, if you update your account through AccountUpdate, you must to sign it with the updated key.

my private key not use AccountUpdate operation.

I read your official docs, that not mention must need  execute AccountUpdate.

but when i follow your step.

```
AccountUpdateTransaction accountUpdateTransaction = AccountUpdateTransaction
        .create(credentials.getAddress(),
            AccountKeyPublic.create(credentials.getEcKeyPair().getPublicKey()),
            BigInteger.valueOf(100_000));
KlayRawTransaction klayRawTransaction = transactionManager.sign(accountUpdateTransaction);
    String send = transactionManager.send(klayRawTransaction);
    System.out.println(send);
```


I use the return value 'send' for sign my transaction, but alse fail.

```
KlayCredentials credentials = KlayCredentials.create("send value");
String rawTransaction = "0x08f87f808505d21dba00830186a094e0711069b41397827fdfc3f31c901018e43015000194c4588d8ca43f4f64e9166f2231ab9aef9dfb3826f847f845824056a019580aa48b23e088e8982ef08e521459b013dc9dd805b5ad7ad1263492eeddaaa030522b7ff369e6b6aaefff0c22032eae4037e024661987ae43cbadec6b7ce700";
```

```
-32000
invalid transaction v, r, s values of the sender
null
```

**Comment by sirano11** [comment]
If you don't want to update your account type, you don't need to executed AccountUpdate Transaction. 

As mentioned above, the "invalid transaction v, r, s values of the sender" error is an error that occurs when you sign with a private key other than the sender's private key.

I decoded your rawTransactions(ValueTransfer, AccountUpdate). It has different "from" (sender) address.
  - ValueTransfer Transaction's sender address("from") = "0x80f4009fb84eb6a151565657fb272bb9f016efb6"
  - AccountUpdate' Transaction's sender address("from") = "0xc4588d8ca43f4f64e9166f2231ab9aef9dfb3826"

It seems that the key you used to create and sign the transaction is not the sender's private key.
Because sender address derived from sender's private key.

Transaction must signed "sender" private key. 
Please double check that the private key used to sign is the sender's private key.

Also checked that the address from your private key(When using signed transaction) and the address of the sender are the same.
```
    KlayCredentials credentials = KlayCredentials.create("your privateKey");
    credentials.getAddress();
```

**Comment by ysykzheng** [comment]
my code:

```
Caver caver = Caver.build(
        new HttpService("https://api.cypress.klaytn.net:8651/", new OkHttpClient.Builder().build(),
            false));
    KlayCredentials credentials = KlayCredentials
        .create("AccountUpdate privateKey");
    TransactionManager transactionManager = new TransactionManager.Builder(caver, credentials)
        .setChaindId(ChainId.MAINNET).build();

    System.out.println(credentials.getAddress());// output 0xc4588d8ca43f4f64e9166f2231ab9aef9dfb3826

    ValueTransferTransaction valueTransferTransaction = ValueTransferTransaction.create(
        credentials.getAddress(),  // fromAddress
        "0xE0711069B41397827fdfc3f31C901018e4301500",  // toAddress
        Convert.toPeb(BigDecimal.ONE, Convert.Unit.PEB).toBigInteger(),  // value
        BigInteger.valueOf(100_000)  // gasLimit
    );

    KlayRawTransaction klayRawTransaction = transactionManager.sign(valueTransferTransaction);
    String rawTransaction = klayRawTransaction.getValueAsString();
    System.out.println(rawTransaction);// output 0x08f87f808505d21dba00830186a094e0711069b41397827fdfc3f31c901018e43015000194c4588d8ca43f4f64e9166f2231ab9aef9dfb3826f847f845824056a019580aa48b23e088e8982ef08e521459b013dc9dd805b5ad7ad1263492eeddaaa030522b7ff369e6b6aaefff0c22032eae4037e024661987ae43cbadec6b7ce700
```

the code output is above. please check? why do you think fault?
if error is mime, please help do better.

**Comment by ysykzheng** [comment]
and the rawTransaction use by original private key, the address is: 0x80f4009fb84eb6a151565657fb272bb9f016efb6

String rawTransaction = "0x08f87f1a8505d21dba00830186a094e0711069b41397827fdfc3f31c901018e4301500019480f4009fb84eb6a151565657fb272bb9f016efb6f847f845824055a0ab0871aced6f1a73b55bf0491cb0039b703eac64dd57745e5404218e64cd6ce3a0588f3150f4096ab697fc59878a084d29599faf94bc72658dd5a15332a83cdb5d";

the error is same.

**Comment by sirano11** [comment]
An addresss - 0x80f4009fb84eb6a151565657fb272bb9f016efb6

Your account has been updated through AccountUpdate transaction. 3 hours ago.
https://scope.klaytn.com/account/0x80f4009fb84eb6a151565657fb272bb9f016efb6?tabId=txList

If you update account through AccountUpdate transaction, you must sign transaction to updated private key 

This info is generated your getAccountKey() RPC API

```
        Caver caver = Caver.build(Caver.MAINNET_URL);
        caver.klay().getAccountKey("0x80f4009fb84eb6a151565657fb272bb9f016efb6", DefaultBlockParameterName.LATEST).send();
```

```json
{
  "jsonrpc": "2.0",
  "id": 0,
  "result": {
    "keyType": 2,
    "key": {
      "x": "0xb47e35593675c108829728f96fffb3d80da56ab144ddecab314466e80f20305f",
      "y": "0xc4a49326c07d493b41219e99d5e627960b7a115328be6560f48e7152899f7408"
    }
  }
}
```

**Comment by ysykzheng** [comment]
At this point, I don't really understand why I need to execute an AccountUpdateTransaction.

Can The transaction be signed directly through the private key?Can you provide a complete sample code?

If an AccountUpdateTransaction operation must be performed, provide a complete sample code that generates the ValueTransferTransaction object and then executes the TransactionManager.send () operation.

**Comment by sirano11** [comment]
It seems to miscommunication between you and me.

This error occurs that private key signed transaction is different with sender private key. So I just asked you, "Have you ever updated your credential through AccountUpdate Transaction?"

You already update account through AccountUpdate. You can create KlayCredentials instance decoupled private key and address.

```java
public void valueTransfer() {
        Caver caver = Caver.build(Caver.MAINNET_URL);
        KlayCredentials credentials = KlayCredentials.create("your updated private key", "0x80f4009fb84eb6a151565657fb272bb9f016efb6");

        TransactionManager transactionManager = new TransactionManager.Builder(caver, credentials)
                .setChaindId(ChainId.MAINNET).build();

        System.out.println(credentials.getAddress());

        ValueTransferTransaction valueTransferTransaction = ValueTransferTransaction.create(
                credentials.getAddress(),  // fromAddress
                "0xE0711069B41397827fdfc3f31C901018e4301500",  // toAddress
                Convert.toPeb(BigDecimal.ONE, Convert.Unit.PEB).toBigInteger(),  // value
                BigInteger.valueOf(100_000)  // gasLimit
        );

        KlayRawTransaction klayRawTransaction = transactionManager.sign(valueTransferTransaction);
        String rawTransaction = klayRawTransaction.getValueAsString();
        System.out.println(rawTransaction);
    }
```

**Comment by ysykzheng** [comment]
use your code, create rawTransaction string, but do `send`, fail.

```
0x08f87f1c8505d21dba00830186a094e0711069b41397827fdfc3f31c901018e4301500019480f4009fb84eb6a151565657fb272bb9f016efb6f847f845824056a02e9a673192ef5bc12008c04348d1eab232448229c128c0e9f2f8d0deb421f6dfa07728b1929758a7e44996542f7634db59774a58a1a934ba566f64b07755663879
```


and I try do AccountUpdate, get a new private key. you can see by this https://scope.klaytn.com/account/0x80f4009fb84eb6a151565657fb272bb9f016efb6?tabId=txList

try do again.
```
0x08f87f1d8505d21dba00830186a094e0711069b41397827fdfc3f31c901018e4301500019480f4009fb84eb6a151565657fb272bb9f016efb6f847f845824055a0f1cf9f623db9c5c302157672bdacf11dba666ceb0e946abd2d5b86e49c81c293a00628fcb990ecfa6ccda2bec348eaa68b08b7670a3ae2323e114b9fc0845d214f
```

the error is same.

**Comment by ysykzheng** [comment]
hello, can i send email to you? with my test private key.

**Comment by ysykzheng** [comment]
hello, can i send email to you 【kale.kim@groundx.xyz】? with my test private key.

**Comment by sirano11** [comment]
You have already confirmed that in version 1.0.2, the transaction was successfully sent to the Klaytn network. 
When you send an email, And it would be nice if you could add some information.
  - What would you like to check?
  - What do you want to do Caver-Java

And the latest version is 1.4.0. Please use Latest version.

**Comment by ysykzheng** [comment]
please update project docs and readme. I read that, get the version 1.0.2

**Comment by ysykzheng** [comment]
use version 1.4.0, the error is same.

**Comment by ysykzheng** [comment]
please check my code.
[caver-test.zip](https://github.com/klaytn/caver-java/files/4876583/caver-test.zip)

**Comment by ysykzheng** [comment]
I want use this lib for company project.

need offline transaction signing.


before we use wallet 1.3.3 version, but wallet not work.
so we upgrade latest wallet 1.4.2 version, need your help to use this lib instead of web3j.

**Comment by sirano11** [comment]
1. Would you explain the wallet you mentioned?
2. Please explain more detail about your code to help you.
  - The test scenario
  - Part of the KlayTest2 class you want to check.

**Comment by ysykzheng** [comment]
old use this wallet version with web3j 
https://docs.klaytn.com/node/download/v1.3.0

current want use this lib, the chain data is syncing
https://docs.klaytn.com/node/download/v1.4.2

the step `signValueTransferTransaction` use step `testAccountUpdateTransaction` return value, do sign ValueTransferTransaction, then get String value, we will store the KlayRawTransaction to database.

the step `sendSignedTransaction` use step `signValueTransferTransaction` return value.
but fail.

please check test `sendSignedTransaction` and `signValueTransferTransaction`.

**Comment by sirano11** [comment]
The private key used Method "signValueTransferTransaction()" is invalid.

It should be replaced with the private key used in testAccountUpdateTransaction().
 I replace the private key and run valueTransfer so it works fine.

Please check this part.

**Comment by ysykzheng** [comment]
if not use AccountUpdateTransaction return value, why I need do this action?

**Comment by ysykzheng** [comment]
before you say:
This error occurs that private key used to sign transaction is different with sender private key. So I just asked you, "Have you ever updated your credential through AccountUpdate Transaction?"

https://github.com/klaytn/caver-java/issues/134#issuecomment-653400076

**Comment by sirano11** [comment]
At first you said, ValueTransfer has failed with error message - invalid transaction v, r, s values of the sender

This error message occurs private key when using signed transaction is different with sender's private key.
I thought 
  - this account may be decoupled account that address and private key is not matched.
(please refer about decouple key pair. https://docs.klaytn.com/klaytn/design/accounts#decoupling-key-pairs-from-addresses)
  - decoupled account can only make AccountUpdate Transaction.
So I asked you, Could you have send AccountTransaction? 

https://github.com/klaytn/caver-java/issues/134#issuecomment-652225748

If you are **curious about AccountUpdate**, please refer to the link below.
https://docs.klaytn.com/bapp/sdk/caver-java/getting-started#account-update

I would appreciate **if you confirm once again that the Private Key you used to sign is the correct Private Key that corresponds to the From Address.**


Please let me know if you don't understand.

**Comment by ysykzheng** [comment]
It seems to miscommunication between you and me.

The cause of the anomaly, I've just discovered. 

I find the error `invalid transaction v, r, s values of the sender` is not your or me.

The cause of the anomaly, I've just discovered. Web3j version conflict.

I just performed the same action in the separate test code I gave you and the project code. I found that the separate test code could normally sign the transaction and send it out, but the generated signature transaction in the project code could not be executed.

The version used in our company's project code is web3j core 3.2.0, while Caver version 1.4.0 relies on version web3j core 
 4.2.0

thanks you reply.
